src/commands/createDeclarativeAgent.ts

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+import { IActionContext } from "@microsoft/vscode-azext-utils";
+import * as vscode from "vscode";
+import { ApiSpecExportResultFormat } from "../azure/ApiCenter/contracts";
+import { TeamsToolkitExtensionId, TeamsToolkitMinimumVersion } from "../constants";
+import { ApiVersionDefinitionTreeItem } from "../tree/ApiVersionDefinitionTreeItem";
+import { UiStrings } from "../uiStrings";
+import { EnsureExtension } from "../utils/ensureExtension";
+import { writeToTemporaryFile } from "../utils/fsUtil";
+import { GeneralUtils } from "../utils/generalUtils";
+import { treeUtils } from "../utils/treeUtils";
+
+export namespace CreateDeclarativeAgent {
+    export async function createDeclarativeAgent(context: IActionContext, node?: ApiVersionDefinitionTreeItem): Promise<void> {
+        if (!node) {
+            node = await treeUtils.getDefinitionTreeNode(context);
+        }
+        if (!node) {
+            return;
```

Comment:
when the node will be undefined, should we throw error or something like to record such telemetry?

Reply:
treeUtils.getDefinitionTreeNode() will handle the cases of user cancel and node not found. So generally, will not reach this case. I think no need to take extrac action here.